### PR TITLE
Add tests, fix CSS highlighting

### PR DESCRIPTION
**Pull request by aomarks**
- Sets up @web/test-runner in a GitHub action
- Integration tests for `<code-sample>`
- Unit tests for `<codemirror-editor>`
- Found and fixed a bug that meant CSS files didn't get highlighted

Fixes https://github.com/PolymerLabs/code-sample-editor/issues/35

### CODE REVIEW

File: src/test/code-sample.test.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
+/**
```
Comment (justinfagnani) [inline src/test/code-sample.test.ts:1]: For consistency, all of our other repos use a `_test.ts` suffix

File: src/test/code-sample.test.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+import {assert} from '@esm-bundle/chai';
+import '../lib/code-sample.js';
+import {CodeSampleElement} from '../lib/code-sample.js';
+import {html, render} from 'lit-html';
+import {UpdatingElement} from 'lit-element';
+import {CodeMirrorEditorElement} from '../lib/codemirror-editor.js';
+import {CodeSampleProjectElement} from '../lib/code-sample-project.js';
+
+describe('code-sample', () => {
```
Comment (justinfagnani) [inline src/test/code-sample.test.ts:23]: Most other places we're using ttd-style test/suite

File: src/test/code-sample.test.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+  afterEach(() => {
+    container.remove();
+  });
+
+  it('is registered', () => {
+    assert.instanceOf(document.createElement('code-sample'), CodeSampleElement);
+  });
+
+  const pierce = async (...selectors: string[]) => {
```
Comment (justinfagnani) [inline src/test/code-sample.test.ts:39]: Nice utility. Does open-wc have something like this?

File: src/test/code-sample.test.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    });
+    // TODO(aomarks) Chromium and Webkit both fire iframe "load" after the
+    // contentDocument has actually loaded, but Firefox fires it before. Why is
+    // that? If not for that, we wouldn't need to poll here.
+    await new Promise((resolve) => {
+      const check = () => {
+        if (iframe.contentDocument?.body?.textContent?.includes(text)) {
+          resolve();
+        } else {
+          setTimeout(check, 10);
```
Comment (justinfagnani) [inline src/test/code-sample.test.ts:73]: should you limit the number of times this will retry?

File: src/test/codemirror-sample.test.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+      for (const child of Array.from(node.childNodes)) {
+        const r = findNodeWithText(child, text);
+        if (r) {
+          return r;
+        }
+      }
+      return undefined;
+    };
+
+    const assertHighlight = async (
```
Comment (justinfagnani) [inline src/test/codemirror-sample.test.ts:90]: Nice basic test here. If this goes any further I'd reach for screenshot tests. I think there's an issue open on WTR for those.

**Review by justinfagnani — APPROVED** [review]

File: src/test/code-sample.test.ts (hunk 1 above)
Comment (aomarks) [reply src/test/code-sample.test.ts]: Done.

File: src/test/code-sample.test.ts (hunk 2 above)
Comment (aomarks) [reply src/test/code-sample.test.ts]: Done.

File: src/test/code-sample.test.ts (hunk 3 above)
Comment (aomarks) [reply src/test/code-sample.test.ts]: I don't see one in https://github.com/open-wc/open-wc/tree/master/packages/testing-helpers

Maybe we should open-source and contribute kevin's shadow-piercer to open-wc or elsewhere.

File: src/test/code-sample.test.ts (hunk 4 above)
Comment (aomarks) [reply src/test/code-sample.test.ts]: Mocha has its own test timeout, so I don't think we need another one. But it does seem like a good idea to kill the polling if the mocha timeout has been reached. Added a variable to stop calling `setTimeout` if `teardown` has been called.

File: src/test/codemirror-sample.test.ts (hunk 5 above)
Comment (aomarks) [reply src/test/codemirror-sample.test.ts]: Agreed.

**Pull request merged**